docs/design/assetgeneration.md

```diff
@@ -16,26 +16,29 @@ An asset is the generic representation of work-item for installer that needs to
 
 The asset would usually follow these steps to generate its output:
 
-1. Load components from disk. (If found, mark them from deletion, we consume our inputs.)
+1. Fetch its parent assets.
 
-2. Load its components from State object given by installer.
+2. Generate the assets either by:
+    *  Using the parent assets
+    *  Loading from on-disk assets
+    *  Loading from state file
 
-3. If both the sources exist,
+3. If any of the parent assets is **dirty** (currently we think all on-disk assets are **dirty**), then use the parent assets to generate and return **dirty**.
 
-    * Both are same: nothing to do use the State object.
+4. If none of the parent assets are **dirty**, but the asset itself is on disk, then use the on-disk asset and return **dirty**.
 
-    * Both differ: maybe try to merge and mark itself **dirty** or error out.
+5. If none of the parent assets or this asset is **dirty**, but the asset is found in the state file, then use the asset from state file and return **NOT dirty**.
 
-4. If only one exists, Move the disk source to State object.
-
-5. If none exists, Use the assets from State object to create the components and store them in the State object.
+6. If none of the parent assets or this asset is **dirty**, neither is the asset found in the state file, then generate the asset using its parent assets and return **NOT dirty**.
```

Comment:
I suggest a re-word to make this sentence a bit more readable.
`If none of the parent assets are **dirty**, this asset is not **dirty**, and this asset is not found in the state file, then ...`